core/types/data_blob.go

```diff
@@ -0,0 +1,332 @@
+package types
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/kzg"
+	"github.com/ethereum/go-ethereum/params"
+	kbls "github.com/kilic/bls12-381"
+	"github.com/protolambda/go-kzg/bls"
+	"github.com/protolambda/ztyp/codec"
+	"github.com/protolambda/ztyp/tree"
+	"io"
+)
+
+const BLOB_COMMITMENT_VERSION_KZG byte = 0x01
+const MAX_TX_WRAP_KZG_COMMITMENTS = 1 << 24
+const LIMIT_BLOBS_PER_TX = 1 << 24
+
+// Compressed BLS12-381 G1 element
+type KZGCommitment [48]byte
+
+func (p *KZGCommitment) Deserialize(dr *codec.DecodingReader) error {
+	if p == nil {
+		return errors.New("nil pubkey")
+	}
+	_, err := dr.Read(p[:])
+	return err
+}
+
+func (p *KZGCommitment) Serialize(w *codec.EncodingWriter) error {
+	return w.Write(p[:])
+}
+
+func (KZGCommitment) ByteLength() uint64 {
+	return 48
+}
+
+func (KZGCommitment) FixedLength() uint64 {
+	return 48
+}
+
+func (p KZGCommitment) HashTreeRoot(hFn tree.HashFn) tree.Root {
+	var a, b tree.Root
+	copy(a[:], p[0:32])
+	copy(b[:], p[32:48])
+	return hFn(a, b)
+}
+
+func (p KZGCommitment) MarshalText() ([]byte, error) {
+	return []byte("0x" + hex.EncodeToString(p[:])), nil
+}
+
+func (p KZGCommitment) String() string {
+	return "0x" + hex.EncodeToString(p[:])
+}
+
+func (p *KZGCommitment) UnmarshalText(text []byte) error {
+	if p == nil {
+		return errors.New("cannot decode into nil KZGCommitment")
+	}
+	if len(text) >= 2 && text[0] == '0' && (text[1] == 'x' || text[1] == 'X') {
+		text = text[2:]
+	}
+	if len(text) != 96 {
+		return fmt.Errorf("unexpected length string '%s'", string(text))
+	}
+	_, err := hex.Decode(p[:], text)
+	return err
+}
+
+func (p *KZGCommitment) Point() (*kbls.PointG1, error) {
+	return kbls.NewG1().FromCompressed(p[:])
+}
+
+func (kzg KZGCommitment) ComputeVersionedHash() common.Hash {
```

Comment:
Code dup with `crypto/kzg/kzg.go:KzgToVersionedHash`

Reply:
Removed the duplicate in kzg package, updated the tests / usage. The versioned-hashes are an ethereum application detail, not really a KZG thing.